docs/proposals/metadata.md

```diff
@@ -0,0 +1,152 @@
+that implements this Service.
+
+Once this Docker image is imported into ImageRepository, the service might be
+referenced by other metadata (SUBSCRIBES, REQUIRES, COMPANIONS, see bellow.)
+
+
+### SUBSCRIBES
+
+This variable contains a list of services that the Docker image can consume.
+The value can be a valid Docker image name or the service name. In order to
+distinguish between them, the proposed format for the value us following:
+
+```
+SUBSCRIBES="service:redis/6379,image:openshift/redis"
+```
+
+In this case, we lookup user ImageRepositories for the Docker image that
+publishes `redis/6379` service. If there is no match, we continue to search for
+the `openshift/redis` image.
+
+The generation tools should automatically generate corresponding Service for
+each defined service.
+
+### REQUIRES
+
+This variable contains a list of services that the Docker image requires in
+order to work properly. The value format is same as for SUBSCRIBES.  Where the
+SUBSCRIBES setup the infrastructure for pub/sub the REQUIRES might indicate that
+the Docker image requires another Docker image in order to work.  For example, a
+'phpmyadmin' requires 'mysql' service to be available, otherwise it won't work
+properly. A special 'backup' Docker image for mysql will require 'mysql' to be
+available within a Pod in order to do database dump from it.
+
+### COMPANIONS
+
+This variable contains a list of services that are companions to the Docker
+image. For example a 'redis-slave' Docker image is a companion for
+'redis-master' Docker image. Where the SUBSCRIBES defines relations to services
+that are required, the companions are optional and the Docker image can work
+without them.
+
+### SECRETS
+
+This variable contains a list of secrets (secret indentifiers) that the Docker
+image requires to work properly. For example the Docker image might require
+Twitter API keys in order to send HTTP requests to the Twitter API. The
+generation tools should check if the required secrets are available and if not,
+it should ask the user to provide them.
+
+An example value can be: `SECRETS=keys/twitter,keys/facebook`
+
+### SINGLETON
+
+This variable indicates if the Docker image is clusterable or not. In case of
+Docker image with standalone 'mysql' server you don't want to have the number of
+replicas in ReplicationController be higher than 1, otherwise you can risk that
+multiple standalone mysql servers will access data on shared volume.
+
+### HOST_VOLUMES
+
+This variable contains a list of Docker volumes that will be host-mounted
+instead of network mounted. These volumes will not be persistent, but they might
+be faster than network volumes.
+
+### URL_MAPPINGS
+
+This variable contains a list of URL mappings. For example, if your 'phpmyadmin'
+is available on '/phpmyadmin' URL, you can override this by specifying a mapping
+in following form: `/phpmyadmin:/`. In this case the traffic will gets routed
+from '/phpmyadmin' to '/'.
```

Comment:
hm, well you still might want to setup mappings at the Router layer, right?  So being able to define those mappings via image metadata makes sense to me so we can auto-generate the right Router definition.